netlib/tcp.py

```diff
@@ -728,7 +731,18 @@ def convert_to_ssl(self, sni=None, alpn_protos=None, **sslctx_kwargs):
 
     def connect(self):
         try:
-            connection = socket.socket(self.address.family, socket.SOCK_STREAM)
+            if not self.connection:
+                connection = socket.socket(self.address.family, socket.SOCK_STREAM)
+            else:
+                connection = self.connection
```

Comment:
Ok, that's a useful feature to have. Can we add a comment indicating that the second case should only happen in this scenario please?